ledger/acctupdates.go

```diff
@@ -1212,88 +1212,53 @@ func (au *accountUpdates) lookupResource(rnd basics.Round, addr basics.Address,
-			needUnlock = false
-		}
-
-		// Look for resources on disk
-		persistedResources, resourceDbRound, err0 := au.accountsq.LookupLimitedResources(addr, basics.CreatableIndex(assetIDGT), limit, basics.AssetCreatable)
-		if err0 != nil {
-			return nil, basics.Round(0), err0
-		}
-
-		if resourceDbRound == currentDbRound || len(persistedResources) == 0 { // db round will return 0 in this case
-			data = make([]ledgercore.AssetResourceWithIDs, 0, len(persistedResources))
-			for _, pd := range persistedResources {
-				ah := pd.Data.GetAssetHolding()
-
-				var arwi ledgercore.AssetResourceWithIDs
-				if !pd.Creator.IsZero() {
-					ap := pd.Data.GetAssetParams()
 
-					arwi = ledgercore.AssetResourceWithIDs{
-						AssetID: basics.AssetIndex(pd.Aidx),
-						Creator: pd.Creator,
+	data = make([]ledgercore.AssetResourceWithIDs, 0, len(persistedResources))
+	for _, pd := range persistedResources {
+		ah := pd.Data.GetAssetHolding()
 
-						AssetResource: ledgercore.AssetResource{
-							AssetHolding: &ah,
-							AssetParams:  &ap,
-						},
-					}
-				} else {
-					arwi = ledgercore.AssetResourceWithIDs{
-						AssetID: basics.AssetIndex(pd.Aidx),
+		var arwi ledgercore.AssetResourceWithIDs
+		if !pd.Creator.IsZero() {
+			ap := pd.Data.GetAssetParams()
 
-						AssetResource: ledgercore.AssetResource{
-							AssetHolding: &ah,
-						},
-					}
-				}
+			arwi = ledgercore.AssetResourceWithIDs{
+				AssetID: basics.AssetIndex(pd.Aidx),
+				Creator: pd.Creator,
 
-				data = append(data, arwi)
+				AssetResource: ledgercore.AssetResource{
+					AssetHolding: &ah,
+					AssetParams:  &ap,
+				},
 			}
-			// We've found all the resources we could find for this address.
-			return data, currentDbRound, nil
-		}
+		} else {
+			arwi = ledgercore.AssetResourceWithIDs{
+				AssetID: basics.AssetIndex(pd.Aidx),
 
-		if synchronized {
-			if resourceDbRound < currentDbRound {
-				au.log.Errorf("accountUpdates.lookupAssetResources: resource database round %d is behind in-memory round %d", resourceDbRound, currentDbRound)
-				return nil, basics.Round(0), &StaleDatabaseRoundError{databaseRound: resourceDbRound, memoryRound: currentDbRound}
-			}
-			au.accountsMu.RLock()
-			needUnlock = true
-			for currentDbRound >= au.cachedDBRound {
-				au.accountsReadCond.Wait()
+				AssetResource: ledgercore.AssetResource{
+					AssetHolding: &ah,
+				},
 			}
-		} else {
-			// in non-sync mode, we don't wait since we already assume that we're synchronized.
-			au.log.Errorf("accountUpdates.lookupAssetResources: database round %d mismatching in-memory round %d", resourceDbRound, currentDbRound)
-			return nil, basics.Round(0), &MismatchingDatabaseRoundError{databaseRound: resourceDbRound, memoryRound: currentDbRound}
 		}
+
+		data = append(data, arwi)
+	}
+	// We've found all the resources we could find for this address.
+	currentDbRound := resourceDbRound
+	// The resourceDbRound will not be set if there are no persisted resources
+	if len(data) == 0 {
```

Comment:
maybe pull it higher to before the loop to exit without an extra `data` allocation?